[Term Entry] Lua Tables `freeze()` (#3129)
Codecademy/docs: content/lua/concepts/tables/terms/freeze/freeze.md

---
Title: 'freeze()'
Description: 'Make table read-only.'
Subjects:
  - 'Programming Language'
  
Tags:
  - 'Tables'
  - 'Functions'
CatalogContent:
  - 'learn-lua'

---

Lua **`freeze()`** function freezes the provided table, and makes it read-only.

## Syntax

```lua
table.freeze(f)
```

The `freeze()` function attempts to freeze a specified non-frozen table named `f`. The function will fail if `f` is not a table or is already frozen.The function will
 Return the expected table `f`.
 This function helps to easily identify immutable functions for correctness purpose.

## Example

To freeze food in your fridge:

```lua
local frozen_table = table.freeze({
    "🍎",
    "🍌",
    "🍇",
    "🍓",
    "🍉"
})
```

> We intiate a table `frozen_table` , then use the `freeze()` to freeze the table contents.
> Now if you run:

Reviewer: ```suggestion

The variable `frozen_table`  is instantiated with the `freeze()` method to make the contents read-only.

Now, if an element is added:
```

```lua

> frozen_table[1] = "🍔" -- This will throw an error

Reviewer: Instead of commenting the output there should be a ```shell block with the error

```

> The error is caused because because the `freeze()` method has made the table `read-only` and we cannot modify its contents.

Reviewer: ```suggestion
The error is thrown because the `freeze()` method has made the table read-only, so the contents cannot be modified.
```
